### PR TITLE
Support <version>-immutable tagged stable images 

### PR DESCRIPTION
**Pull request by cevich**
Fixes: https://github.com/containers/automation_images/issues/310

There are circumstances where it's consequential to have image digests
that change constantly.  Add a "flavor" of podman/buildah/skopeo image
with a unique tag, that's only ever pushed once.  Update documentation.

### CODE REVIEW

**Comment by cevich** [comment]
@edsantiago it's probably asking a lot, so PLMK if it's asking too much for you to review this.  Esp. the "Simplify image-build automation scripts" commit is quite big since it prunes out a lot of legacy crap.  I did perform a few runs of CI against just that commit, and they're happy.

**Comment by cevich** [comment]
@TomSweeneyRedHat would you mind taking a look at the (short) README updates here, LMK if there are any major problems?

File: .cirrus.yml [hunk 1]
```diff
@@ -6,9 +6,7 @@ env:
     CIRRUS_SHELL: "/bin/bash"
     # No need to go crazy, but grab enough to cover most PRs
     CIRRUS_CLONE_DEPTH: 10
-    # Prefix git URL for every container tool's repo home
-    REPO_PREFIX: https://github.com/containers
-    IMAGE_SUFFIX: "c20240320t153921z-f39f38d13"
+    IMAGE_SUFFIX: "c20240411t124913z-f39f38d13"
```
Comment (edsantiago) [inline .cirrus.yml:9]: DANGER! `c20240411t124913z` is from https://github.com/containers/automation_images/pull/340 which is my `/var/tmp` change which has not been approved. Until (unless) that is approved and distributed to the other tools, that ID should be considered toxic.

Comment (cevich) [reply .cirrus.yml:9]: Noted.  I simply grabbed it because it was recent.  I'll fine another one that was merged.

**Comment by cevich** [comment]
@edsantiago NP at all if this is too much, I appreciate you're making an effort.

File: README.md [hunk 2]
```diff
@@ -23,10 +23,15 @@ The container images are tagged as follows, where `*` represents either `podman`
 or `skopeo`:
 
   * `quay.io/containers/*:<version>` and `quay.io/*/stable:<version>` -
-    These images are built daily.  They are intended to contain an unchanging
-    and stable version of their container image. For the most recent `<version>` tags (`vX`,
+    These images are built daily.  They are intended to contain the latest stable
+    versions of their container tool. For the most recent `<version>` tags (`vX`,
```
Comment (TomSweeneyRedHat) [inline README.md:27]: A very soft suggestion for consideration

```suggestion
versions of their respective container tool. For the most recent `<version>` tags (`vX`,
```

File: ci/containers_build_push.sh [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+# the build as a `--build-arg`.  This is used by the `Containerfile` to alter
+# the build to produce a 'stable', 'testing', or 'upstream' image.  Importantly,
+# this also determines where the image is pushed, see the top-level README.md for
+# more details.
+#
+# Optionally, the `$ARCHES` environment variable may be set to a comma-separated
+# list of golang-centric architectures to include in the build.  It is assumed
+# that the necessary emulation is setup to handle building of non-native arches.
+# Note: these builds will run in parallel, which can make the output difficult
+# to read.
```
Comment (TomSweeneyRedHat) [inline ci/containers_build_push.sh:28]: Great description here, but no metnion of immutable, it might be nice to add.

**Comment by TomSweeneyRedHat** [comment]
Couple doc changes for considration or to be ignored.  Your call.
LGTM otherwise.

File: ci/containers_build_push.sh (hunk 3 above)
Comment (cevich) [reply ci/containers_build_push.sh:28]: good catch, I'll mention it.

**Comment by cevich** [comment]
This is ready to go, letting it sit for a day or so in case there are last minute objections/feedback.

**Comment by cevich** [comment]
Note to me: Check next cron run and verify immutable images have been pushed.  Send mail to podman list assuming images aren't overwritten on the subsequent cron run.

**Pull request merged**